java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -2281,6 +2281,30 @@ public final ColumnVector repeatStrings(int repeatTimes) {
     return new ColumnVector(repeatStrings(getNativeView(), repeatTimes));
   }
 
+  /**
+   * Given a strings column, each string in the given column is repeated a number of times
+   * given by the corresponding row in a <code>repeatTimes</code> numeric column. If any row in
+   * that column contains a non-positive value and the corresponding input string is not null, the
+   * output string for that row will always be an empty string. Any null row (from either the input
+   * strings column or the <code>repeatTimes</code> column) will result in a null output.
+   *
+   * If the input <code>repeatTimes</code> column does not have a numeric type, an exception will be
+   * thrown.
+   *
+   * Note that this function cannot handle the cases when the size of the output column exceeds
+   * the maximum value that can be indexed by int type (i.e., {@link Integer#MAX_VALUE}).
+   * In such situations, the output result is undefined.
```

Comment:
How is the RAPIDS Accelerator going to use this if the behavior is undefined when the total number of characters in the resulting string column exceeds 2GB?  It seems like it would need to compute this separately to know whether it can safely call this function.  I would expect the libcudf function to check for this overflow when it computes the new offset column.

Reply:
I implemented that overflow checking, but was rejected because "that is the job of the caller". So, we have to do checking on the plug in side.

Reply:
So is there going to be a libcudf utility method to help check for this overflow in this context?  With the data being on the GPU, I don't see how the plugin side is going to be able to check for the possibility of overflow in a performant manner before calling this method without help from libcudf.  It would essentially need to repeat (no pun intended) the calculation being performed here, at least for the offset vector of the result.

Note that `cudf::strings::detail::gather` checks for offset vector overflow in the result, so I'm surprised a method like this doesn't similarly check.

Reply:
Yeah, this is a problem. We may need to do bounds check for similar string processing situations as well. I'm filing an issue for this.

Reply:
Alright, I'm trying to add overflow checking to this new API, hoping that it will not be rejected again.

Reply:
I think a benchmark for this method would help decide what to do about checks.  That way we can see the impact of the check and determine whether it's worth worrying about whether and how to help the caller perform the checks separately.